eventuate-messaging-kafka-basic-consumer/src/main/java/io/eventuate/messaging/kafka/basic/consumer/SaveOffsetOnRebalanceListener.java

```diff
@@ -0,0 +1,37 @@
+package io.eventuate.messaging.kafka.basic.consumer;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+
+public class SaveOffsetOnRebalanceListener implements ConsumerRebalanceListener {
```

Comment:
This class is for a custom implementation. It's not actually used by default and so does not belong in this repository.